moped-editor/src/views/projects/projectView/ProjectActivityEntry.js

```diff
@@ -10,22 +10,27 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-const ProjectActivityEntry = ({
-  changeIcon,
-  changeDescription,
-  changeValue,
-}) => {
+/**
+ * One row/entry in the project activity log
+ * @param {changeIcon} Material UI icon component
+ * @param {changeText} - Array of objects of the shape:
+ *  {text: String to display, style: String name of style or null}
+ */
+const ProjectActivityEntry = ({ changeIcon, changeText }) => {
   const classes = useStyles();
-
   return (
     <Box display="flex" p={0}>
       <Box p={0}>{changeIcon}</Box>
       <Box p={0} flexGrow={1}>
         <Typography variant="body2" className={classes.entryText}>
-          {changeDescription}
-          {!!changeValue && (
-            <span className={classes.boldText}>{changeValue}</span>
-          )}
+          {
+            // maps through the array of objects and applies specified style to the text
+            changeText.map((changeObject) => (
+              <span className={classes[changeObject.style]}>
```

Comment:
would it be worth adding a conditional to only add a className if the style prop exists?

Reply:
good question, i thought if the classname was `null` it would ignore it but i am going to make sure its null and not `undefined`

Reply:
thank you for checking :)